### PR TITLE
OpenSSF scorecard report

### PR DESCRIPTION
**Pull request by vjuntunen**
Add OpenSSF scorecard github action

## Description of changes

Add OpenSSF Scorecard github action and scorecard badge to README which also operates as a link to scorecard report.

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `nix flake check --accept-flake-config` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status

## Testing

Tested scorecard functionality in github action and compared to command line output to verify results.
Scorecard badge works on README.md in my fork.

### CODE REVIEW

**Comment by vilvo** [comment]
Please remove "DRAFT" from the PR header and check "Author has added reviewers and removed PR draft status"

**Review by vilvo — APPROVED** [review]

**Comment by henrirosten** [comment]
Reviewing, do not merge yet.

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# SPDX-FileCopyrightText: 2022-2023 TII (SSRC) and the Ghaf contributors
+#
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:2]: What is the origin of this file? We should probably add another copyright line, respecting the original copyright owner as well as the license from the origin.

File: .github/workflows/scorecard.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '0 0 8 * * '
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:17]: Why `'0 0 8 * * '` ? That's once a month every eight day of the month at midnight.
The example [here](https://github.com/ossf/scorecard-action?tab=readme-ov-file#workflow-example) is weekly, we should at least explain why we don't use the weekly schedule.

File: .github/workflows/scorecard.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:39]: Consider updating to latest release version: https://github.com/actions/checkout/releases

File: .github/workflows/scorecard.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86 # v2.1.2
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:44]: Consider updating to latest release version: https://github.com/ossf/scorecard-action/releases

File: .github/workflows/scorecard.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          #   - See https://github.com/ossf/scorecard-action#publishing-results.
+          # For private repositories:
+          #   - `publish_results` will always be set to `false`, regardless
+          #     of the value entered here.
+          publish_results: true
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:66]: Consider updating to latest release version: https://github.com/actions/upload-artifact/releases

File: .github/workflows/scorecard.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        with:
+          name: SARIF file
+          path: results.sarif
+          retention-days: 5
+
+      # Upload the results to GitHub's code scanning dashboard.
+      - name: "Upload to code-scanning"
+        uses: github/codeql-action/upload-sarif@17573ee1cc1b9d061760f3a006fc4aac4f944fd5 # v2.2.4
```
Comment (henrirosten) [inline .github/workflows/scorecard.yml:74]: Consider updating to latest release version: https://github.com/github/codeql-action/releases

File: README.md [hunk 7]
```diff
@@ -19,6 +19,9 @@ This repository contains the source files (code and documentation) of Ghaf Frame
 
 For information on build instructions and supported hardware, see the [Reference Implementations](https://tiiuae.github.io/ghaf/ref_impl/reference_implementations.html) section of Ghaf documentation.
 
+### OpenSSF Scorecard
+
+[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/vjuntunen/ghaf-scorecard/badge)](https://securityscorecards.dev/viewer/?uri=github.com/vjuntunen/ghaf-scorecard)
```
Comment (henrirosten) [inline README.md:24]: Based on https://github.com/ossf/scorecard-action?tab=readme-ov-file#scorecard-badge, the Badge and here point to incorrect repo. This points to @vjuntunen 's fork, which I believe is not correct.

This line should be change to:
`[OpenSSF Scorecard](https://securityscorecards.dev/viewer/?uri=github.com/tiiuae/ghaf)`

**Review by henrirosten — CHANGES_REQUESTED** [review]

File: .github/workflows/scorecard.yml (hunk 1 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml:2]: https://github.com/marketplace/actions/ossf-scorecard-action workflow example/default scorecard.yml

File: .github/workflows/scorecard.yml (hunk 2 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml]: Updated running weekly on Sundays

File: .github/workflows/scorecard.yml (hunk 3 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml]: Updated

File: .github/workflows/scorecard.yml (hunk 4 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml]: Updated

File: .github/workflows/scorecard.yml (hunk 5 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml]: Updated

File: .github/workflows/scorecard.yml (hunk 6 above)
Comment (vjuntunen) [reply .github/workflows/scorecard.yml]: Updated

File: README.md (hunk 7 above)
Comment (vjuntunen) [reply README.md]: fixed

**Review by henrirosten — APPROVED** [review]

**Review by vilvo — APPROVED** [review]

**Pull request merged**